spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/reactive/PubSubReactiveFactory.java

```diff
@@ -0,0 +1,168 @@
+	 * with remaining demand being fulfilled in the future.
+	 * Pub/Sub timeout will cause a retry with the same demand.
+	 * @param subscriptionName subscription from which to retrieve messages.
+	 * @param pollingPeriodMs how frequently to poll the source subscription in case of unlimited demand, in milliseconds.
+	 * @return infinite stream of {@link AcknowledgeablePubsubMessage} objects.
+	 */
+	public Flux<AcknowledgeablePubsubMessage> createPolledFlux(String subscriptionName, long pollingPeriodMs) {
+
+		return Flux.create(sink -> {
+			sink.onRequest((numRequested) -> {
+				if (numRequested == Long.MAX_VALUE) {
+					// unlimited demand
+					Disposable task = Schedulers.single().schedulePeriodically(
+							new PubSubNonBlockingUnlimitedDemandPullTask(subscriptionName, sink), 0, pollingPeriodMs, TimeUnit.MILLISECONDS);
+					sink.onCancel(task::dispose);
+				}
+				else {
+					Schedulers.single().schedule(new PubSubBlockingLimitedDemandPullTask(subscriptionName, numRequested, sink));
+				}
+			});
+
+		});
+	}
+
+	private abstract class PubSubPullTask implements Runnable {
+
+		protected final String subscriptionName;
+
+		protected final FluxSink<AcknowledgeablePubsubMessage> sink;
+
+		PubSubPullTask(String subscriptionName, FluxSink<AcknowledgeablePubsubMessage> sink) {
+			this.subscriptionName = subscriptionName;
+			this.sink = sink;
+		}
+
+		/**
+		 * Retrieve up to a specified number of messages, sending them to the subscription.
+		 * @param demand maximum number of messages to retrieve
+		 * @param block whether to wait for the first message to become available
+		 * @return number of messages retrieved
+		 */
+		protected int pullToSink(int demand, boolean block) {
+
+			List<AcknowledgeablePubsubMessage> messages =
+					PubSubReactiveFactory.this.subscriberOperations.pull(this.subscriptionName, demand, !block);
+
+			if (!this.sink.isCancelled()) {
+				messages.forEach(sink::next);
+			}
+
+			return messages.size();
+		}
+
+	}
+
+	/**
+	 * Runnable task issuing blocking Pub/Sub Pull requests until the specified number of
+	 * messages has been retrieved.
+	 */
+	private class PubSubBlockingLimitedDemandPullTask extends PubSubPullTask {
+
+		private final long initialDemand;
+
+		PubSubBlockingLimitedDemandPullTask(String subscriptionName, long initialDemand, FluxSink<AcknowledgeablePubsubMessage> sink) {
+			super(subscriptionName, sink);
+			this.initialDemand = initialDemand;
+		}
+
+		@Override
+		public void run() {
+			long demand = this.initialDemand;
+			List<AcknowledgeablePubsubMessage> messages;
+
+			while (demand > 0 && !this.sink.isCancelled()) {
+				try {
+					int intDemand = demand > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) demand;
+					demand -= pullToSink(intDemand, true);
+				}
+				catch (DeadlineExceededException e) {
+					LOGGER.trace("Blocking pull timed out due to empty subscription " + this.subscriptionName + "; retrying.");
```

Comment:
add isTraceEnabled()

Reply:
No reason: this is in the `catch()`  block, so only in case of exception.
It's not going to be a big deal to build a logging message in case of this exception.

Reply:
This exception is actually expected. So, it may happen quiet often.

Reply:
Every 10 minutes by default, which is negligible in computer speed. On the other hand, we might as well add `isTraceEnabled()` for consistency.